docs/mount-change-detection.md

```diff
@@ -0,0 +1,58 @@
+# Mount-point change detection
+
+## Introduction
+
+This document presents the design of the mount-point change detection system in NDM.
+The goal of mount-point change detection is to detect the changes in the mount-points
+and the filesystem on the existing blockdevices discovered by NDM and trigger appropriate
+action to update the blockdevice CRs.
+
+## Design
+
+The mount-points and filesystem info of a block device are found by the probe - *mountprobe*.
```

Comment:
I checked the `findmnt` manual page and it says it looks for mount points in `/etc/fstab` ,  `/etc/mtab` and  `/proc/self/mountinfo` . Just wondering if there's a reason NDM should look at fstab or mtab too.

Reply:
Not sure if we should be watching fstab/mtab. I took a look at `findmnt` though. It provies an option to watch for changes (`-p`). To implement this, it also [uses the linux polling api](https://github.com/karelzak/util-linux/blob/059ba3dd76c9d6cf45ce01e01bb03091a14a48a9/misc-utils/findmnt.c#L1145) and by default wacthes for changed in `/proc/self/mountinfo`.

Reply:
Cool, that's good validation. We can add watching fstab/mtab if the need ever arises I suppose. Right now, don't see a reason we need it